src/Components/Components/src/Routing/Router.cs

```diff
@@ -45,6 +46,10 @@ public partial class Router : IComponent, IHandleAfterRender, IDisposable
 
     [Inject] private ILoggerFactory LoggerFactory { get; set; }
 
+    [Inject] IServiceProvider ServiceProvider { get; set; }
+
+    private IRoutingStateProvider RoutingStateProvider { get; set; }
```

Comment:
```suggestion
    private IRoutingStateProvider? RoutingStateProvider { get; set; }
```

Right? Why is the build not complaining about this?

And nit: should we also move this up above the normally-`[Inject]`ed properties now?

Reply:
> Why is the build not complaining about this?

https://github.com/dotnet/aspnetcore/blob/4afe7f612d104b43b690e71d83c18a8bc48aae2d/src/Components/Components/src/Routing/Router.cs#L4

Reply:
If we use `[Inject]` then it won't be an optional service.